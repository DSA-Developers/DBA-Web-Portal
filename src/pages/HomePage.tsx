import React from 'react';
import styled from 'styled-components';
import Container from 'react-bootstrap/Container';
import { CenterRow } from '../components/CenterRow';
import Button from 'react-bootstrap/Button';

interface Props {}

const Title = styled.h1`
  font-family: Roboto;
`;

export const HomePage: React.FC<Props> = ({}) => {
  return (
    <Container>
      <CenterRow>
        <Title>Home Page</Title>
      </CenterRow>
      <CenterRow>
        <h2>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi fermentum elementum lorem. Quisque ac dapibus lacus, in sagittis leo. Praesent fermentum nec massa et imperdiet. Fusce sapien nisl, lacinia ut ante eu, laoreet elementum ex. Duis eget purus a dui pellentesque maximus. Aenean vel ultricies velit. Mauris scelerisque sapien dolor, a laoreet magna aliquet ut. Vestibulum nec dignissim tellus.
        Recusandae autem et ipsam. Saepe qui et repellat. Maxime distinctio non occaecati cum voluptates voluptatibus quo minima. Inventore facilis a accusantium quaerat ut quam quos. Enim qui porro nobis in qui ratione.
        Porro non maxime et porro veniam ut laudantium. Magni quidem non officiis eaque nobis quidem. Laboriosam doloremque quos eius. Maiores quia est explicabo nisi consectetur. Consectetur velit ducimus maiores iusto quia natus aliquid. Odio libero magnam praesentium accusantium pariatur quia perspiciatis incidunt.
        Provident consectetur quo nostrum veniam. Nam inventore ullam repellat soluta corrupti quam. Dolor quis aliquam exercitationem rerum iure placeat voluptatem.
        Id non rem reprehenderit dolor similique et. Et ut laborum velit incidunt. Ea excepturi et repudiandae. Rem culpa rerum dolores accusantium sed magnam facilis officia.
        Quaerat exercitationem minima dolor amet rerum nesciunt est occaecati. Labore quidem quae excepturi. Et praesentium quidem nihil quos.
        </h2>
      </CenterRow>
    </Container>
  );
};
