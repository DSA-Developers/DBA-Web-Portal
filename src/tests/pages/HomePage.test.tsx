import renderer from 'react-test-renderer';
import HomePage from '../../pages/HomePage';

test('Home Page renders', () => {
  const props = {};
  const component = renderer.create(<HomePage {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
