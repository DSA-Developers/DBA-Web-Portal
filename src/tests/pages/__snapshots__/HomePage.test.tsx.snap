// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders 1`] = `
.c16 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  width: 300px;
}

.c16:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c17 {
  width: 150px;
  height: 150px;
  border-radius: 25px;
  margin-right: 100px;
}

.c18 {
  font-family: Optima,Segoe,'Segoe UI',Candara,Calibri,Arial,sans-serif;
  font-size: 22px;
  font-style: normal;
  font-weight: 450;
  line-height: 26.4px;
  padding: 5%;
  text-align: center;
}

.c2 {
  padding: 20px 400px 20px 400px;
  background-color: #f8f9fb;
  width: 100vw;
  textalign: 'center';
}

.c3 {
  font-family: Optima,Segoe,'Segoe UI',Candara,Calibri,Arial,sans-serif;
  font-size: 50px;
  font-style: normal;
  font-variant: small-caps;
  font-weight: 700;
  padding: 4%;
  margin-bottom: -2%;
  text-align: center;
  background: -webkit-linear-gradient(0.5turn,#ce1126,#002d62);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.c4 {
  font-family: Optima,Segoe,'Segoe UI',Candara,Calibri,Arial,sans-serif;
  font-size: 35px;
  font-style: normal;
  font-variant: normal;
  font-weight: 350;
  line-height: 60px;
  padding: 5% 10%;
  text-align: center;
}

.c14 {
  font-family: Optima,Segoe,'Segoe UI',Candara,Calibri,Arial,sans-serif;
  font-size: 27px;
  font-style: normal;
  font-variant: small-caps;
  font-weight: 600;
  line-height: 26.4px;
  padding: 5%;
  text-align: center;
}

.c7 Button.[disabled] {
  background-color: #17a2b8;
}

.c8 {
  font-family: Optima,Segoe,'Segoe UI',Candara,Calibri,Arial,sans-serif;
  font-size: 35px;
  font-style: normal;
  font-variant: normal;
  font-weight: 400;
}

.c9 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  max-width: 250px;
}

.c9:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c6 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px 40px 20px 0px;
}

.c10 {
  width: 100px;
  height: 100px;
  border-radius: 25px;
}

.c12 {
  padding: 20px 0px 20px 0px;
  background-color: #f8f9fb;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: grid;
  margin: 50px 0px;
}

.c11 {
  padding: 20px 200px 50px 200px;
  background-color: #f8f9fb;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  padding: 0px 0px 0px 0px;
  margin-bottom: -40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  padding: 20px 0px 20px 0px;
  background-color: #fff;
  width: 100vw;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 50px -13px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-template-columns: repeat(3,150px);
  grid-gap: 30px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 15px;
}

.c15 {
  margin: 25px;
  display: grid;
  grid-template-columns: repeat(6,140px);
  box-sizing: border-box;
  width: 1400px;
  height: 10%;
  -webkit-column-gap: 100px;
  column-gap: 100px;
  text-align: center;
}

.c19 {
  display: grid;
  grid-template-columns: repeat(3,140px);
  margin-right: 70px;
  -webkit-column-gap: 100px;
  column-gap: 100px;
  text-align: center;
}

.c1 {
  width: 100vw;
  height: auto;
}

<div
  className="container-fluid"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="c0"
  >
    <div
      className="carousel slide"
      onKeyDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <ol
        className="carousel-indicators"
      >
        <li
          className="active"
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
      </ol>
      <div
        className="carousel-inner"
      >
        <div
          className="active carousel-item"
          interval={3500}
        >
          <img
            className="c1"
            src="DSA Carousel.jpg"
          />
        </div>
        <div
          className="carousel-item"
          interval={3500}
        >
          <img
            className="c1"
            src="DSA Carousel2.jpg"
          />
        </div>
        <div
          className="carousel-item"
          interval={3500}
        >
          <img
            className="c1"
            src="DSA Carousel3.jpg"
          />
        </div>
      </div>
      <a
        className="carousel-control-prev"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
        <span
          className="sr-only"
        >
          Previous
        </span>
      </a>
      <a
        className="carousel-control-next"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
        <span
          className="sr-only"
        >
          Next
        </span>
      </a>
    </div>
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      Welcome to the Dominican Student Association at UF!
    </div>
    <h1
      className="c4"
    >
      ¡¿Que lo que?! Welcome to the Dominican Student Association at the University of Florida! Join our GroupMe or follow us on Instagram or Facebook for event information, memes, and friendship!
    </h1>
  </div>
  <div
    className="c5"
  >
    <p
      className="c6"
    >
      <button
        className="c7 btn btn-info"
        disabled={true}
        type="button"
      >
        <h1
          className="c8"
        >
          Join Now
        </h1>
      </button>
    </p>
    <div
      className="c9"
    >
      <a
        href="https://www.facebook.com/DSAatUF"
        target="_blank"
        title="Facebook"
      >
        <img
          className="c10"
          src="facebook.svg"
        />
      </a>
    </div>
    <div
      className="c9"
    >
      <a
        href="https://www.instagram.com/uf.dsa/"
        target="_blank"
        title="Instagram"
      >
        <img
          className="c10"
          src="instagram.svg"
        />
      </a>
    </div>
    <div
      className="c9"
    >
      <a
        href="https://groupme.com/join_group/53205471/hXrFzndj"
        target="_blank"
        title="GroupMe"
      >
        <img
          className="c10"
          src="groupme.svg"
        />
      </a>
    </div>
  </div>
  <div
    className="c11"
  >
    <div
      className="c3"
    >
      Check us out on Instagram!
    </div>
    <br />
    <div
      className="powr-instagram-feed"
      id="2c6cfa4f_1617477419"
    />
  </div>
  <div
    className="c12"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
  >
    <div
      className="c3"
    >
      Parent Organization
    </div>
    <br />
    <div
      className="c13"
    >
      <div
        className="c9"
      >
        <a
          href="https://www.ufhsa.com/"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            src="HispanicStudentLogo.png"
            style={
              Object {
                "paddingLeft": "30px",
              }
            }
          />
          <div
            className="card-title h5"
            style={
              Object {
                "paddingTop": "20px",
              }
            }
          >
            <h1
              className="c14"
            >
              Hispanic Student Association
            </h1>
          </div>
        </a>
      </div>
    </div>
    <br />
    <div
      className="c3"
    >
      <h1>
        In Association With:
      </h1>
    </div>
    <div
      className="c15"
    >
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/argentina"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="ArgentineStudentsLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Argentine Student Association
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/tango"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="ArgentineTango.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Argentine Tango Club
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/cuba"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="CubanAmericanLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Cuban American Student Association
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/centralamerica"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="CentralAmericanLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Central American Latino Organization
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/salsa"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="GatorSalsaLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Gator Salsa Club
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/hispanic-communicators"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="HispanicCommunicatorsLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Hispanic Communicators Association
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/mexico"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="MexicanStudentLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Mexican-American Student Association
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/peru"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="PeruvianAmericanLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Peruvian American Student Society
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/colombia"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="PorColombiaLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Por Colombia
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/sabor-latino"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="SaborLatinoLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Sabor Latino
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/puertorico"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="UnionDeEstudiantesLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              La Union de Estudiantes Puertorriqueños Activos
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/venezuela"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="VenezuelanStudentLogo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Venezuelan Student Association
            </h1>
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="c12"
    style={
      Object {
        "backgroundColor": "#f8f9fb",
        "marginBottom": "0px",
      }
    }
  >
    <div
      className="c3"
    >
      General Resources
    </div>
    <br />
    <div
      className="c19"
    >
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/campus-resources"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="UF logo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Campus Resources
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/hsa-resources"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="HSA Logo.png"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              HSA Resources
            </h1>
          </div>
        </a>
      </div>
      <div
        className="c16"
      >
        <a
          href="https://www.ufhsa.com/mental-health-resources"
          style={
            Object {
              "color": "#000",
            }
          }
          target="_blank"
        >
          <img
            className="c17"
            src="GatorWell.jpg"
          />
          <div
            className="card-title h5"
            style={
              Object {
                "fontSize": "22px",
                "padding": "25px 100px 25px 0px",
                "textAlign": "center",
              }
            }
          >
            <h1
              className="c18"
            >
              Mental Health
            </h1>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
